src/common/cockpitpipe.c

```diff
@@ -1505,6 +1529,47 @@ cockpit_pipe_get_stderr (CockpitPipe *self)
   return priv->err_buffer;
 }
 
+gchar *
+cockpit_pipe_take_stderr_as_utf8 (CockpitPipe *self)
+{
+  GByteArray *buffer;
+  GBytes *bytes;
+  GBytes *clean;
+  gchar *data;
+  gsize length;
+
+  buffer = cockpit_pipe_get_stderr (self);
+  if (!buffer)
+    return NULL;
+
+  /* A little more complicated to avoid big copies */
+  g_byte_array_ref (buffer);
+  g_byte_array_append (buffer, (guint8 *)"x", 1); /* place holder for null terminate */
+  bytes = g_byte_array_free_to_bytes (buffer);
+  clean = cockpit_unicode_force_utf8 (bytes);
+  g_bytes_unref (bytes);
+
+  data = g_bytes_unref_to_data (clean, &length);
+
+  /* Fill in null terminate, for x above */
```

Comment:
Probably this would all be a lot cleaner if you just passed the buffer, unmodified, into the force function, got the byte array back out, and added the nul after...

Are we expecting such large strings here?

Reply:
`cockpit_unicode_force_utf8` returns a GBytes object which doesn't offer any way to append to it, no?  We could append a `\0` instead of a "x" and change cockpit_unicode_force_utf8 to accept it.

Reply:
Even though it's not included in the length of the `GBytes`, the string from `cockpit_unicode_force_utf8()` comes from `GString` and is therefore already always nul terminated.

Maybe add a doc comment to the top of `cockpit_unicode_force_utf8()` mentioning this fact, and remove all the extra hacks here.